xbmc/input/InputManager.cpp

```diff
@@ -458,11 +458,30 @@ bool CInputManager::OnEvent(XBMC_Event& newEvent)
   switch (newEvent.type)
   {
   case XBMC_KEYDOWN:
+  {
     m_Keyboard.ProcessKeyDown(newEvent.key.keysym);
-    OnKey(m_Keyboard.TranslateKey(newEvent.key.keysym));
+    CKey key = m_Keyboard.TranslateKey(newEvent.key.keysym);
+    if (!CButtonTranslator::GetInstance().HasLonpressMapping(g_windowManager.GetActiveWindowID(), key))
+    {
+      m_LastKey.Reset();
+      OnKey(key);
+    }
+    else
+    {
+      if (key.GetButtonCode() != m_LastKey.GetButtonCode() && key.GetButtonCode() & CKey::MODIFIER_LONG)
+      {
+        m_LastKey = key;  // OnKey is reentrant; need to do this before entering
+        OnKey(key);
```

Comment:
@koying long pressing enter on a DialogKeyboard results in an ```ACTION_ENTER``` there: https://github.com/xbmc/xbmc/pull/7346/files#diff-82f619a2901d7602884fec1217d18283R1239
And then we hit https://github.com/xbmc/xbmc/blob/master/xbmc/dialogs/GUIDialogKeyboardGeneric.cpp#L184
. Is this intended?